Sources/BSONCore/BinaryDecodableValue.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  BinaryDecodableValue.swift
+//
+//
+//  Created by Christopher Richez on September 27th, 2021
+//
+
+/// A binary value that follows BSON binary type rules.
+///
+/// Conforming to this protocol instead of `TopLevelDecodableValue` provides
+/// default implementations for the `encodedSize(in:)` and `findValueSubsequence(in:)`
+/// methods, as well as a new `subtype()` method that returns the declared subtype byte.
+///
+/// Conformance to `ReadableFromByteArray` is still required. You need to provide
+/// the `init(bytes:)` initializer where `bytes` is the binary data excluding the subtype
+/// and size bytes. Type information is managed by the decoder.
+///
+/// ```swift
+/// extension MyCustomType: BinaryDecodableValue, TopLevelDecodableValue, ReadableFromByteArray {
+///     public init<T>(bytes: T) throws where T : Sequence, T.Element == UInt8 {
+///         <#code#>
+///     }
+/// }
+/// ```
+public protocol BinaryDecodableValue: TopLevelDecodableValue {}
+
+extension BinaryDecodableValue {
+    public static func encodedSize<T>(in remainingDocument: T) throws -> Int
+    where T : RandomAccessCollection, T.Element == UInt8, T.Index == Int {
+        let size = try Int32(bytes: remainingDocument.prefix(4))
+        return Int(size) + 5
```

Comment:
Add a comment explaining the +5 since it isn't immediately obvious.
```suggestion
        // Return the declared size + 5 for the size and subtype bytes
        return Int(size) + 5
```